test/test.js

```diff
@@ -3301,6 +3303,30 @@ describe('Page', function() {
       }
     });
   });
+
+  describe('Response', function() {
+    it('specifies when response comes from cache and service worker', async() => {
+      const browser = await puppeteer.launch(defaultBrowserOptions);
+      const newPage = await browser.newPage();
+
+      // Use known site that uses server caching and service worker caching.
+      let responses = [];
+      newPage.on('response', response => {
+        if (response.url.includes('.js'))
+          responses.push(response);
+      });
+
+      await newPage.goto('https://www.chromestatus.com/features', {waitUntil: 'networkidle2'});
```

Comment:
please don't use external URLs; we aim for the hermetic builds.

Reply:
> I tried to use one-style.html for the test but it would require additions to sw.js and the test server doesn't support caching AFAIK.

What do you suggest for the server part?

Reply:
I've added caching to the test server. Still seeing a lot of unrelated errors that I could use your help on. Not familiar enough with the worker and custom test server.